### PR TITLE
Add tasks widget

### PR DESCRIPTION
**Pull request by AzmiTouil**

### CODE REVIEW

File: task-management/src/main/webapp/css/main.less [hunk 1]
```diff
@@ -0,0 +1,377 @@
+// Variables
+@primaryColor: #578DC9;
+@greyColor: #a8b3c5;
+@yellowColor: #ffc20e;
+@blueGreyColor: #4d5466;
+@greyBackground: #e1e8ee;
+@btnPrimaryDisableBackground: #afc9e5;
+
+/*-------------------------Global page Container--------------------------*/
```
Comment (azayati) [inline task-management/src/main/webapp/css/main.less:9]: Should be removed

File: task-management/src/main/webapp/css/main.less [hunk 2]
```diff
@@ -0,0 +1,377 @@
+// Variables
+@primaryColor: #578DC9;
```
Comment (azayati) [inline task-management/src/main/webapp/css/main.less:2]: Keep only used variables

File: task-management/src/main/webapp/css/main.less [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+@yellowColor: #ffc20e;
+@blueGreyColor: #4d5466;
+@greyBackground: #e1e8ee;
+@btnPrimaryDisableBackground: #afc9e5;
+
+/*-------------------------Global page Container--------------------------*/
+
+/*--------------------------Tasks ---------------------------*/
+
+#digital-workplace-tasks {
```
Comment (azayati) [inline task-management/src/main/webapp/css/main.less:13]:
```suggestion
#tasks {
```

File: task-management/src/main/webapp/css/main.less [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+        color: #a8b3c5;
+        margin-top: -50px;
+      }
+    }
+  }
+}
+
+/*----------- Task Drawer CSS ---------------*/
+
+#digital-workplace-tasks {
```
Comment (azayati) [inline task-management/src/main/webapp/css/main.less:70]:
```suggestion
#tasks {
```

File: task-management/src/main/webapp/vue-app/tasks/main.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import tasksApp from './components/TasksApp.vue';
+import TaskDrawer from './components/TaskDrawer.vue';
+import TaskDetails from './components/TaskDetails.vue';
+import TaskLabels from './components/TaskLabels.vue';
+import TaskProjects from './components/TaskProject.vue';
+import TaskAssignment from './components/TaskAssignment.vue';
+
+import '../../css/main.less';
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/tasks/main.js:8]: This should be removed

File: task-management/webpack.common.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  // can use multiple entry
+  entry: {
+    tasks: './src/main/webapp/vue-app/tasks/main.js',
+  },
+  output: {
+    filename: 'js/[name].bundle.js',
+    libraryTarget: 'amd'
+  },
+  module: {
+    rules: [
```
Comment (azayati) [inline task-management/webpack.common.js:16]: Normally the css plugin should be removed and replaced with portlet skin in gatein -resources

File: task-management/webpack.dev.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+const path = require('path');
+const merge = require('webpack-merge');
+const webpackCommonConfig = require('./webpack.common.js');
+
+// the display name of the war
+const app = 'task-management';
+
+// add the server path to your server location path
+const exoServerPath = "/home/exo/Digitale-Workplace/platform-6.0.x-SNAPSHOT/";
```
Comment (azayati) [inline task-management/webpack.dev.js:9]: this 
```suggestion
const exoServerPath = "/exo-server";
```

File: task-management/src/main/webapp/css/main.less (hunk 1 above)
Comment (AzmiTouil) [reply task-management/src/main/webapp/css/main.less]: Done

File: task-management/src/main/webapp/css/main.less (hunk 2 above)
Comment (AzmiTouil) [reply task-management/src/main/webapp/css/main.less]: Done

File: task-management/src/main/webapp/vue-app/tasks/main.js (hunk 5 above)
Comment (AzmiTouil) [reply task-management/src/main/webapp/vue-app/tasks/main.js]: Done

File: task-management/webpack.common.js (hunk 6 above)
Comment (AzmiTouil) [reply task-management/webpack.common.js:16]: Done

File: task-management/src/main/webapp/skin/css/tasks.less [hunk 8]
```diff
@@ -0,0 +1,539 @@
+// Variables
+@primaryColor: #578DC9;
+@greyColor: #a8b3c5;
+@yellowColor: #ffc20e;
```
Comment (azayati) [inline task-management/src/main/webapp/skin/css/tasks.less:4]: No used, remove it

File: task-management/src/main/webapp/skin/css/tasks.less [hunk 9]
```diff
@@ -0,0 +1,539 @@
...
+@blueGreyColor: #4d5466;
```
Comment (azayati) [inline task-management/src/main/webapp/skin/css/tasks.less:5]: No used, remove it

File: task-management/src/main/webapp/vue-app/tasks/main.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+const vuetify = new Vuetify({
+    dark: true,
+    iconfont: '',
+});
+
+// getting language of user
+const lang = eXo && eXo.env && eXo.env.portal && eXo.env.portal.language || 'en';
+
+const resourceBundleName = 'locale.portlet.dw.Portlets';
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/tasks/main.js:15]: This should be the tasks bundle not dw one

File: task-management/src/main/webapp/vue-app/tasks/components/TasksApp.vue [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+            mx-0
+            align-center>
+            <v-flex
+              d-flex
+              xs12>
+              <v-card
+                flat
+                color="transparent">
+                <a @click="navigateTo('tasks/my-task')"><v-card-text class="body-1 text-uppercase color-title px-0">
+                  {{ this.$t('label.tasks.header') }}
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/tasks/components/TasksApp.vue:26]:
```suggestion
                  {{ $t('label.tasks.header') }}
```

File: task-management/src/main/webapp/vue-app/tasks/components/TasksApp.vue [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+                              </v-flex>
+                              <v-flex
+                                d-flex
+                                xs8>
+                                <v-card
+                                  class="py-1"
+                                  flat
+                                  color="#F7FAFD"
+                                  @click="navigateTo('tasks')">
+                                  <div class="title">{{ incomingTasksSize }} {{ this.$t('label.tasks.tasks') }}</div>
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/tasks/components/TasksApp.vue:82]:
```suggestion
                                  <div class="title">{{ incomingTasksSize }} {{ $t('label.tasks.tasks') }}</div>
```

File: task-management/src/main/webapp/vue-app/tasks/components/TasksApp.vue [hunk 13]
```diff
@@ -0,0 +1,239 @@
...
+                                  <div class="caption color-title">{{ this.$t('label.incoming') }}</div>
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/tasks/components/TasksApp.vue:83]:
```suggestion
                                  <div class="caption color-title">{{ $t('label.incoming') }}</div>
```

File: task-management/src/main/webapp/vue-app/tasks/components/TasksApp.vue [hunk 14]
```diff
@@ -0,0 +1,239 @@
...
+                              </v-flex>
+                              <v-flex
+                                d-flex
+                                xs7>
+                                <v-card
+                                  class="py-1"
+                                  flat
+                                  color="#F7FAFD"
+                                  @click="navigateTo('tasks/my-task/overdue')">
+                                  <div class="title">{{ overdueTasksSize }} {{ this.$t('label.tasks.tasks') }}</div>
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/tasks/components/TasksApp.vue:134]:
```suggestion
                                  <div class="title">{{ overdueTasksSize }} {{ $t('label.tasks.tasks') }}</div>
```

File: task-management/src/main/webapp/vue-app/tasks/components/TasksApp.vue [hunk 15]
```diff
@@ -0,0 +1,239 @@
...
+                                  <div class="caption color-title">{{ this.$t('label.overdue') }}</div>
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/tasks/components/TasksApp.vue:135]:
```suggestion
                                  <div class="caption color-title">{{ $t('label.overdue') }}</div>
```

File: task-management/src/main/webapp/vue-app/tasks/components/TasksApp.vue [hunk 16]
```diff
@@ -0,0 +1,239 @@
...
+                  v-for="task in tasks"
+                  :key="task.id"
+                  class="px-0">
+                  <task-details :task="task"/>
+                </v-list-item>
+              </v-list>
+              <div v-else class="noTasks">
+                <div class="noTasksContent">
+                  <i class="uiNoTaskIcon"></i>
+                  <p class="pt-4 noTasksTitle">{{ this.$t('label.noTask') }}</p>
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/tasks/components/TasksApp.vue:169]:
```suggestion
                  <p class="pt-4 noTasksTitle">{{ $t('label.noTask') }}</p>
```

File: task-management/src/main/webapp/vue-app/tasks/components/TasksApp.vue [hunk 17]
```diff
@@ -0,0 +1,239 @@
...
+                  <p class="pt-10">{{ this.$t('label.thereIsNoTasks') }}</p>
```
Comment (azayati) [inline task-management/src/main/webapp/vue-app/tasks/components/TasksApp.vue:170]:
```suggestion
                  <p class="pt-10">{{ $t('label.thereIsNoTasks') }}</p>
```

**Pull request merged**